packages/storybook/stories/MessageThread/MessageThread.stories.tsx

```diff
@@ -97,7 +97,70 @@ const mentionTag = `
   Displayable Text
 </msft-mention>
 `;
-const getDocs: () => JSX.Element = () => {
+const Docs: () => JSX.Element = () => {
+  const refDefaultMessageThread = useRef(null);
+  const refWithMessageDate = useRef(null);
+  const refSystemMessage = useRef(null);
+  const refBlockedMessage = useRef(null);
+  const refCustomMessage = useRef(null);
+  const refWithCustomizedChatContainer = useRef(null);
+  const refWithCustomizedMessageContainer = useRef(null);
+  const refWithCustomizedBlockedMessageContainer = useRef(null);
+  const refDefaultMessageWithStatusIndicator = useRef(null);
+  const refCustomMessageWithStatusIndicator = useRef(null);
+  const refCustomAvatar = useRef(null);
+  const refCustomTimestamp = useRef(null);
+  const refDisplayInlineImages = useRef(null);
+  const refDisplayFileAttachments = useRef(null);
+  const refMentionOfUsers = useRef(null);
+  const refProps = useRef(null);
+
+  const scrollToRef = (ref): void => {
+    ref.current.scrollIntoView({ behavior: 'auto' });
+  };
+
+  useEffect(() => {
+    const url = window.top ? window.top.location.href : window.location.href;
+    if (url.includes('default-messagethread') && refDefaultMessageThread.current) {
+      scrollToRef(refDefaultMessageThread);
+    } else if (url.includes('messagethread-with-message-date') && refWithMessageDate.current) {
+      scrollToRef(refWithMessageDate);
+    } else if (url.includes('system-message') && refSystemMessage.current) {
+      scrollToRef(refSystemMessage);
+    } else if (url.includes('blocked-message') && refBlockedMessage.current) {
```

Comment:
In the URL, both `refWithCustomizedBlockedMessageContainer` and `refBlockedMessage` have the substring `blocked-message`. If you attempt to scroll to `refWithCustomizedBlockedMessageContainer` using the URL `https://60c7ae6891f0e90039d7cd54-jojdfeztms.chromatic.com/?path=/docs/ui-components-messagethread--message-thread#messages-with-customized-blocked-message-container`, it will fail. This is because `refBlockedMessage` matches `blocked-message` first and thus, it scrolls to `refBlockedMessage` before anything else.

To address this, we need to prioritize the more specific path, `messages-with-customized-blocked-message-container`, over the general `blocked-message` path. The revised order should be:

```javascript
if (
  url.includes('messages-with-customized-blocked-message-container') &&
  refWithCustomizedBlockedMessageContainer.current
) {
  scrollToRef(refWithCustomizedBlockedMessageContainer);
} else if (url.includes('blocked-message') && refBlockedMessage.current) {
  scrollToRef(refBlockedMessage);
}
A similar approach may be needed for custom-message.

Reply:
Oh good catch. Yes I based it off of order of item. There might be a case as well for simplicity to group them both under the same header. But I will put in our solution for now